pkg/operator/starter.go

```diff
@@ -40,6 +46,94 @@ const (
 	resyncInterval = 20 * time.Minute
 )
 
+
+func addDCObjectHash(deployment *appsv1.Deployment, inputHashes map[string]string) error {
+	if deployment == nil {
+		return fmt.Errorf("invalid deployment: %v", deployment)
+	}
+	if deployment.Annotations == nil {
+		deployment.Annotations = map[string]string{}
+	}
+	if deployment.Spec.Template.Annotations == nil {
+		deployment.Spec.Template.Annotations = map[string]string{}
+	}
+	for k, v := range inputHashes {
+		annotationKey := fmt.Sprintf("operator.openshift.io/dep-%s", k)
+		if len(annotationKey) > 63 {
+			hash := sha256.Sum256([]byte(k))
+			annotationKey = fmt.Sprintf("operator.openshift.io/dep-%x", hash)
+			annotationKey = annotationKey[:63]
+		}
+		deployment.Annotations[annotationKey] = v
+		deployment.Spec.Template.Annotations[annotationKey] = v
+	}
+	return nil
+}
+
+// TODO(stephenfin): This should be migrated to library-go
+// WithDCConfigMapHashAnnotationHook creates a deployment hook that annotates a Deployment with a config map's hash.
+func WithDCConfigMapHashAnnotationHook(
+	namespace string,
+	configMapName string,
+	configMapInformer corev1.ConfigMapInformer,
+) dc.DeploymentHookFunc {
+	return func(opSpec *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		inputHashes, err := resourcehash.MultipleObjectHashStringMapForObjectReferenceFromLister(
+			configMapInformer.Lister(),
+			nil,
+			resourcehash.NewObjectRef().ForConfigMap().InNamespace(namespace).Named(configMapName),
+		)
+		if err != nil {
+			return fmt.Errorf("invalid dependency reference: %w", err)
+		}
+		return addDCObjectHash(deployment, inputHashes)
+	}
+}
+
+func addDSObjectHash(daemonSet *appsv1.DaemonSet, inputHashes map[string]string) error {
+	if daemonSet == nil {
+		return fmt.Errorf("invalid daemonSet: %v", daemonSet)
+	}
+	if daemonSet.Annotations == nil {
+		daemonSet.Annotations = map[string]string{}
+	}
+	if daemonSet.Spec.Template.Annotations == nil {
+		daemonSet.Spec.Template.Annotations = map[string]string{}
+	}
+	for k, v := range inputHashes {
+		annotationKey := fmt.Sprintf("operator.openshift.io/dep-%s", k)
+		if len(annotationKey) > 63 {
+			hash := sha256.Sum256([]byte(k))
+			annotationKey = fmt.Sprintf("operator.openshift.io/dep-%x", hash)
+			annotationKey = annotationKey[:63]
+		}
+		daemonSet.Annotations[annotationKey] = v
+		daemonSet.Spec.Template.Annotations[annotationKey] = v
+	}
+	return nil
+}
+
+// TODO(stephenfin): This should be migrated to library-go
+// WithDSConfigMapHashAnnotationHook creates a DaemonSet hook that annotates a DaemonSet with a config map's hash.
+func WithDSConfigMapHashAnnotationHook(
```

Comment:
Do we really need to implement the same change for the DaemonSet? The nodeplugins don't look at the topology key. Or do we need it for reacting to cloud.conf changes?

Reply:
It's for the latter. We want to ensure all pods are respun whenever the config map changes.